sn_networking/src/event/kad.rs

```diff
@@ -402,9 +407,40 @@ impl SwarmDriver {
                     Self::send_record_after_checking_target(sender, peer_record.record, &cfg)?;
                 } else {
                     debug!("For record {pretty_key:?} task {query_id:?}, fetch completed with split record");
-                    sender
-                        .send(Err(GetRecordError::SplitRecord { result_map }))
-                        .map_err(|_| NetworkError::InternalMsgChannelDropped)?;
+                    let mut accumulated_spends = BTreeSet::new();
+                    for (record, _) in result_map.values() {
+                        match get_raw_signed_spends_from_record(record) {
+                            Ok(spends) => {
+                                accumulated_spends.extend(spends);
+                            }
+                            Err(_) => {
+                                break;
+                            }
+                        }
+                    }
+                    if !accumulated_spends.is_empty() {
+                        info!("For record {pretty_key:?} task {query_id:?}, found split record for a spend, accumulated and sending them as a single record");
+                        let accumulated_spends = accumulated_spends
+                            .into_iter()
+                            .take(2)
```

Comment:
why `2` ? why not more?

Reply:
Can we keep a random magic number here for now, say 50? If so, might have to do the same during PUT validation (we store 2 there).

The lower bound of a single signed spend is 825B. Can grow if the inputs/outputs increase.